tests/test_packages/test_cleveref.py

```diff
@@ -0,0 +1,72 @@
+from yalafi import parameters, parser, utils
+
+
+latex_1 = r"""
+\usepackage[poorman]{cleveref}
+\newcommand{\YYCleverefInput}[1]{}  % this definition should be ignored
+\YYCleverefInput{latex.sed}
+\cref{eq:1}
+"""
+sed_1 = r"""
+s/\\cref{eq:1}/\\cref@equation@name \\nobreakspace \\textup {(\\ref {eq:1})}/g
+s/\\cref@equation@name /eq\./g
+"""
+plain_1 = """
+eq.\N{NO-BREAK SPACE}(0)
+"""
+
+
+def test_1():
+    def read(file):
+        return True, sed_1
+    parms = parameters.Parameters()
+    p = parser.Parser(parms, read_macros=read)
+    toks = p.parse(latex_1)
+    plain, pos = utils.get_txt_pos(toks)
+    assert plain_1 == plain
+
+
+latex_2 = r"""
+\usepackage[poorman]{cleveref}
+\YYCleverefInput{latex.sed}
+\Cref*{eq:1}
+"""
+sed_2 = r"""
+s/\\Cref\*{eq:1}/\\Cref@equation@name \\nobreakspace \\textup {(\\ref \*{eq:1})}/g
+s/\\Cref@equation@name /Equation/g
+"""
+plain_2 = """
+Equation\N{NO-BREAK SPACE}(0)
+"""
+
+
+def test_2():
+    def read(file):
+        return True, sed_2
+    parms = parameters.Parameters()
+    p = parser.Parser(parms, read_macros=read)
+    toks = p.parse(latex_2)
+    plain, pos = utils.get_txt_pos(toks)
+    assert plain_2 == plain
+
+
+latex_3 = r"""
+\usepackage[poorman]{cleveref}
+\YYCleverefInput{latex.sed}
+\crefrange{eq:1}{eq:3}
+"""
+sed_3 = r"""
+s/\\crefrange{eq:1}{eq:3}//g
+"""
+plain_3 = """
+"""
+
+
+def test_3():
+    def read(file):
+        return True, sed_3
+    parms = parameters.Parameters()
+    p = parser.Parser(parms, read_macros=read)
+    toks = p.parse(latex_3)
+    plain, pos = utils.get_txt_pos(toks)
+    assert plain_3 == plain
```

Comment:
While working on the fix for the file recursion problem in issue #169, I was thinking about checking error messages sent to `stderr`. Here is a possible application:
```
latex_4 = r"""
\usepackage[poorman]{cleveref}
A\cref{lab}B
"""
plain_4 = """
A LATEXXXERROR B
"""

stderr_4 = r"""*** LaTeX error: line 3, column 2:
*** To use cleveref with YaLafi, you should use \YYCleverefInput to load
*** the sed file, e.g. '\YYCleverefInput{main.sed}' if your LaTeX
*** document is called 'main.tex'.

"""

def test_4(capsys):
    capsys.readouterr()
    parms = parameters.Parameters()
    p = parser.Parser(parms)
    toks = p.parse(latex_4)
    plain, pos = utils.get_txt_pos(toks)
    captured = capsys.readouterr()
    assert plain_4 == plain
    assert stderr_4 == captured.err
```